rpc/src/cors.rs

```diff
@@ -0,0 +1,46 @@
+use hyper::{http::HeaderValue, Method};
+use tower_http::cors::{AllowOrigin, Any, CorsLayer};
+
+/// Error thrown when parsing cors domains went wrong
+#[derive(Debug, thiserror::Error)]
+pub(crate) enum CorsDomainError {
+    #[error("{domain} is an invalid header value")]
+    InvalidHeader { domain: String },
+    #[error("Wildcard origin (`*`) cannot be passed as part of a list: {input}")]
+    WildCardNotAllowed { input: String },
+}
+
+/// Creates a [CorsLayer] from the given domains
+pub(crate) fn create_cors_layer(http_cors_domains: &str) -> Result<CorsLayer, CorsDomainError> {
```

Comment:
What is the use case where we would need support for cors?

Reply:
Enabling CORS in Trin allows web applications hosted on specified domains to make HTTP requests to the Trin HTTP-RPC server and interact with the Portal network. 

Almost all of the Ethereum clients implement CORS which I think is good to have it also for Trin.